lib/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+cmake_minimum_required(VERSION 3.22...3.25)
+
+legacy_check()
+
+add_library(obs-websocket-api INTERFACE)
+add_library(OBS::websocket-api ALIAS obs-websocket-api)
+
+target_sources(obs-websocket-api INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/obs-websocket-api.h>
+                                           $<INSTALL_INTERFACE:${OBS_INCLUDE_DESTINATION}/obs-websocker-api.h>)
+
+target_link_libraries(obs-websocket-api INTERFACE OBS::libobs)
+
+target_include_directories(obs-websocket-api INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>"
+                                                       "$<INSTALL_INTERFACE:${OBS_INCLUDE_DESTINATION}>")
+
+set_target_properties(
+  obs-websocket-api
+  PROPERTIES FOLDER websocket
```

Comment:
The `FOLDER` property for the API as well as obs-websocket itself needs to be adjusted to say the same (`plugins/obs-websocket`) to group them.

Reply:
I made big changes, so please re-review if those properties are correctly set.